core-web/libs/edit-content/src/lib/feature/edit-content/edit-content.layout.component.spec.ts

```diff
@@ -0,0 +1,270 @@
+            required: false,
+            searchable: false,
+            sortOrder: 5,
+            unique: false,
+            variable: 'text3'
+        }
+    ],
+    fixed: false,
+    folder: 'SYSTEM_FOLDER',
+    host: '48190c8c-42c4-46af-8d1a-0cd5db894797',
+    iDate: 1697051073000,
+    icon: 'event_note',
+    id: 'd46d6404125ac27e6ab68fad09266241',
+    layout: LAYOUT_MOCK,
+    modDate: 1697051118000,
+    multilingualable: false,
+    name: 'Test-K',
+    system: false,
+    systemActionMappings: {},
+    variable: 'TestK',
+    versionable: true,
+    workflows: [
+        {
+            archived: false,
+            creationDate: new Date(1697047303976),
+            defaultScheme: false,
+            description: '',
+            entryActionId: null,
+            id: 'd61a59e1-a49c-46f2-a929-db2b4bfa88b2',
+            mandatory: false,
+            modDate: new Date(1697047292887),
+            name: 'System Workflow',
+            system: true
+        }
+    ],
+    nEntries: 0
+};
+
+const createEditContentLayoutComponent = (params: { contentType: string; id: string }) => {
+    return createComponentFactory({
+        component: EditContentLayoutComponent,
+        imports: [HttpClientTestingModule],
+        componentProviders: [
+            {
+                provide: ActivatedRoute,
+                useValue: { snapshot: { params } }
+            }
+        ]
+    });
+};
+
+describe('EditContentLayoutComponent with identifier', () => {
+    let spectator: Spectator<EditContentLayoutComponent>;
+    let dotEditContentService: SpyObject<DotEditContentService>;
+
+    const createComponent = createEditContentLayoutComponent({ contentType: 'test', id: '1' });
+
+    beforeEach(() => {
+        spectator = createComponent({
+            detectChanges: false,
+            providers: [
+                {
+                    provide: DotEditContentService,
+                    useValue: {
+                        getContentTypeFormData: jest.fn().mockReturnValue(of(LAYOUT_MOCK)),
+                        getContentById: jest.fn().mockReturnValue(of(CONTENT_TYPE_MOCK)),
+                        saveContentlet: jest.fn().mockReturnValue(of({}))
+                    }
+                }
+            ]
+        });
+
+        dotEditContentService = spectator.inject(DotEditContentService, true);
+    });
+
+    it('should set contentType and identifier from activatedRoute', () => {
+        expect(spectator.component.contentType).toEqual('test');
+        expect(spectator.component.identifier).toEqual('1');
+    });
+
+    it('should call getContentById and getContentTypeFormData with contentType if identifier is present', () => {
+        spectator.detectChanges();
+
+        expect(dotEditContentService.getContentById).toHaveBeenCalledWith('1');
+    });
+
+    it('should call dotEditContentService.saveContentlet with the correct parameters', () => {
+        spectator.component.saveContent({ key: 'value' });
```

Comment:
If I remove the event from the HTML, this test will keep passing, but the functionality will be broken.

Reply:
👆🏽